src/InstrumentationEngine.Lib/FileLoggerSink.cpp

```diff
@@ -75,45 +75,55 @@ HRESULT CFileLoggerSink::Reset(_In_ LoggingFlags defaultFlags, _Out_ LoggingFlag
             wszExtension,
             _MAX_EXT);
 
-        tstring tsFile(wszFile);
-        tstring tsExtension(wszExtension);
+        // Try to create log directory if it doesn't exist
+        BOOL fDirectoryExists = PathFileExists(wszDirectory);
+        if (!fDirectoryExists)
+        {
+            fDirectoryExists = CreateDirectory(wszDirectory, nullptr);
```

Comment:
It looks like CreateDirectory can return 0, with an ERROR_ALREADY_EXISTS error code. CreateDirectory also doesn't handle intermediate paths. You probably want to check both of those error codes.